### PR TITLE
Token minting

### PR DESCRIPTION
**Pull request by feritka79**
Initial Project Folder Structure
Database Configuration and Token Mode
Creating the .env and .env.example Files
Installing the body-parser Package

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,12 +1,25 @@
...
-  res.send("dkfdhfkdhk");
-});
+//connect database
+db();
+
+//env file
+dotenv.config();
+
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
```
Comment (ceydae) [inline src/index.ts:18]: Change these to `bodyParser.json` and `bodyParser.urlencoded`

File: src/database/db.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import mongoose from 'mongoose';
+const MONGO_URI = 'mongodb://localhost:27017/Fluxity';
```
Comment (ceydae) [inline src/database/db.ts:2]: use process.env and put it in env and env.example

File: src/database/db.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+const db = async () => {
+  try {
+    await mongoose.connect(MONGO_URI);
+  } catch (error) {
+    console.log(error);
+  }
```
Comment (ceydae) [inline src/database/db.ts:9]: Change the error text to something meaningful
exit the app

File: src/database/db.ts
Comment (ceydae) [inline src/database/db.ts:1]: move db.ts to src/ and remove database directory

File: src/index.ts [hunk 4]
```diff
@@ -1,12 +1,25 @@
...
 const port = 3000;
 
-app.get("/a", (req, res) => {
-  res.send("dkfdhfkdhk");
-});
+//connect database
+db();
+
+//env file
+dotenv.config();
```
Comment (ceydae) [inline src/index.ts:15]: move this up, below the imports

File: src/index.ts [hunk 5]
```diff
@@ -1,12 +1,25 @@
...
+import indexHandler from './routes/token/indexHandler';
 
 const app = express();
 const port = 3000;
 
-app.get("/a", (req, res) => {
-  res.send("dkfdhfkdhk");
-});
+//connect database
+db();
```
Comment (ceydae) [inline src/index.ts:12]: pass process.env.DB_URL to this as an argument

File: src/index.ts [hunk 6]
```diff
@@ -1,12 +1,25 @@
...
-import express from "express";
+import express from 'express';
+import db from './database/db';
+import dotenv from 'dotenv';
+import bodyParser from 'body-parser';
+
+import indexHandler from './routes/token/indexHandler';
 
 const app = express();
 const port = 3000;
```
Comment (ceydae) [inline src/index.ts:9]: put it in env and env.example

File: src/models/tokens.ts [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import mongoose from 'mongoose';
+
+const Tokens = new mongoose.Schema({
+  address: { type: String, required: true, unique: true },
+  symbol: { type: String, required: true, unique: true },
+});
+
+export default mongoose.model('Tokens', Tokens);
```
Comment (ceydae) [inline src/models/tokens.ts:8]: use Token instead of Tokens

File: src/index.ts [hunk 8]
```diff
@@ -1,12 +1,25 @@
...
+//connect database
+db();
+
+//env file
+dotenv.config();
+
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
+
+app.get('/', indexHandler);
```
Comment (ceydae) [inline src/index.ts:20]: move this to routes/index.ts

**Review by ceydae — CHANGES_REQUESTED** [review]

**Review by feritka79 — COMMENTED** [review]
fix

**Review by ceydae — CHANGES_REQUESTED** [review]
routes

**Review by feritka79 — COMMENTED** [review]
add router

**Review by feritka79 — COMMENTED** [review]
fix route index and add folder token (index, addToken)

**Review by feritka79 — COMMENTED** [review]
change import name

**Review by ceydae — APPROVED** [review]

**Pull request merged**